caddy.go

```diff
@@ -877,3 +877,7 @@ type CtxKey string
 // context.WithValue to access the original request URI that accompanied the
 // server request. The associated value will be of type string.
 const URLPathCtxKey CtxKey = "url_path"
+
+// URIxRewriteCtxKey is a context key used to store original unrewritten
+// URI in context.WithValue
+const URIxRewriteCtxKey CtxKey = "caddy_rewrite_original_uri"
```

Comment:
Nuts. This goes back to the debate I had when I designed Caddy 0.9 about whether the staticfiles package should be in the httpserver package as one package. It's kind of an edge case because the staticfiles is technically a middleware but it's the "base" middleware that is always used with the httpserver. So, uh... I guess that's one fail of my decision earlier.

Sigh. Let's leave the consts where they are, but put a TODO in a comment saying we want them in the httpserver package eventually, but we haven't done that yet since it creates a circular import with the staticfiles package. Then we can merge.